### PR TITLE
Kavit's changes of QR detection on Priyams skeleton code

### PR DESCRIPTION
**Pull request by KavitShah1998**
@zephirefaith please review this PR for changes.

### CODE REVIEW

File: aria_data_loaders/aria_data_utils/aria_live_streamer.py [hunk 1]
```diff
@@ -134,20 +148,73 @@ def disconnect(self):
...
+
+        Args:
+            image (np.ndarray): Image to be rectified or undistorted
+
+        Returns:
+            np.ndarray: Rectified image
+        """
+        assert self._rgb_calib_params is not None and self._dst_calib_params is not None
+        print(self._dst_calib_params)
+        print(self._rgb_calib_params)
```
Comment (zephirefaith) [inline aria_data_loaders/aria_data_utils/aria_live_streamer.py:165]: rospy.logdebug

File: aria_data_loaders/aria_data_utils/aria_live_streamer.py [hunk 2]
```diff
@@ -134,20 +148,73 @@ def disconnect(self):
...
         Process the frame
         """
-        output = {}
+
+        # Initialize camera_T_marker to None & object_scores to empty dict for current image frame
+        camera_T_marker = None
+        object_scores = {}
+
+        # Rectify current image frame
+        frame["image"] = self._rectify_image(image=frame.get("image"))
```
Comment (zephirefaith) [inline aria_data_loaders/aria_data_utils/aria_live_streamer.py:198]: take a look at the new code in aria_live_streamer. frame has "raw_image" which is only rectified and a "image" which is the rect+rot one.

Could you update your code to play friendly with that?

File: aria_data_loaders/aria_data_utils/aria_live_streamer.py [hunk 3]
```diff
@@ -134,20 +148,73 @@ def disconnect(self):
...
+
         if detect_qr:
-            output.update(self.april_tag_detector.process_frame(frame))
+            (frame["image"], camera_T_marker) = self.april_tag_detector.process_frame(img_frame=frame.get("image"))  # type: ignore
```
Comment (zephirefaith) [inline aria_data_loaders/aria_data_utils/aria_live_streamer.py:201]: Why are we updating "image"? Output should not be overwriting the input, right? Let's add it to a new key "qr_output".

File: aria_data_loaders/aria_data_utils/detector_wrappers/april_tag_detector.py [hunk 4]
```diff
@@ -134,17 +133,18 @@ def get_outputs(
         """
         # Decorate image with text for visualization
         img = decorate_img_with_text(
-            img=img_frame,
+            img=img_frame.get("image"),
             frame_name="device",
             position=device_T_marker.translation(),
         )
+        timestamp = img_frame.get("timestamp")
         print(
```
Comment (zephirefaith) [inline aria_data_loaders/aria_data_utils/detector_wrappers/april_tag_detector.py:141]: I'd prefer rospy.logdebug every where you have debugging prints currently.

File: aria_data_loaders/aria_data_utils/aria_live_streamer.py [hunk 5]
```diff
@@ -184,12 +252,21 @@ def main(do_update_iptables: bool):
...
-            _ = aria_object.process_frame(frame)
+        if frame is not None:  # During init, self._latest_frame was {} which caused code to come in this block
+            frame["image"], outputs = aria_object.process_frame(frame, outputs=outputs,detect_qr=True, detect_hand_object=False)
+            # if aria_object.aria_rgb_frame is not None:
+            image = frame.get("image")
+            # else:
+            #     image = np.zeros((1408, 1408, 3))
+
+            cv2.imshow("Image", image[:, :, ::-1])
+            cv2.waitKey(1)
```
Comment (zephirefaith) [inline aria_data_loaders/aria_data_utils/aria_live_streamer.py:269]: Don't overwrite input "image". Change this when you fix that.

File: aria_data_loaders/aria_data_utils/perception/april_tag_pose_estimator.py [hunk 6]
```diff
@@ -117,13 +114,8 @@ def detect_markers_and_estimate_pose(
...
-        mn_camera_T_marker = Spot.convert_transformation_from_sophus_to_magnum(
-            sp_camera_T_marker
-        )
 
         if should_render:
             image = self._cam_module.render_markers(image, markers=[marker])
 
-        if magnum:
-            return image, mn_camera_T_marker
         return image, sp_camera_T_marker
```
Comment (zephirefaith) [inline aria_data_loaders/aria_data_utils/perception/april_tag_pose_estimator.py:121]: is there no averaging ATM?

**Review by zephirefaith — CHANGES_REQUESTED** [review]
Thanks for this PR, Kavit! Nothing major, few stylistic and readability comments. Let's have this play nice with the current `aria_live_streamer`.

File: aria_data_loaders/aria_data_utils/perception/april_tag_pose_estimator.py (hunk 6 above)
Comment (KavitShah1998) [reply aria_data_loaders/aria_data_utils/perception/april_tag_pose_estimator.py:121]: No.

File: aria_data_loaders/aria_data_utils/detector_wrappers/april_tag_detector.py (hunk 4 above)
Comment (KavitShah1998) [reply aria_data_loaders/aria_data_utils/detector_wrappers/april_tag_detector.py:142]: +1

File: aria_data_loaders/aria_data_utils/aria_live_streamer.py (hunk 1 above)
Comment (KavitShah1998) [reply aria_data_loaders/aria_data_utils/aria_live_streamer.py:196]: its useless. removed

File: aria_data_loaders/aria_data_utils/detector_wrappers/april_tag_detector.py (hunk 4 above)
Comment (KavitShah1998) [reply aria_data_loaders/aria_data_utils/detector_wrappers/april_tag_detector.py:142]: addressed

**Review by zephirefaith — APPROVED** [review]

**Pull request merged**